update_scripts/update_existing_samples.py

```diff
@@ -165,10 +165,11 @@ def process_request(self, campaign_uid, request):
             root_request = self.get_request_if_exists(root_request_prepid)
             miniaod_request = self.get_request_if_exists(miniaod_prepid)
             nanoaod_request = self.get_request_if_exists(nanoaod_prepid)
-
+            other_tags = sorted_join(request['tags'])
```

Comment:
This is completely missing the bidirectionality of GrASP and McM updates. `other_tags = sorted_json(request['tags'])` will always overwrite any changes in GrASP with tags in McM. Check the `process_interested_pwgs_update` method.